ios/RCTGoogleAnalyticsBridge/RCTGoogleAnalyticsBridge/RCTGoogleAnalyticsBridge.m

```diff
@@ -138,6 +138,14 @@ - (NSDictionary *)constantsToExport
                                                         target:targetUrl] build]];
 }
 
+RCT_EXPORT_METHOD(setCustomDimensionValue:(NSString *)trackerId dimensionIndex:(NSInteger)dimensionIndex dimensionValue:(NSString *)dimensionValue)
+{
+    id<GAITracker> tracker = [[GAI sharedInstance] trackerWithTrackingId:trackerId];
+    
+    [tracker set:[GAIFields customDimensionForIndex:dimensionIndex] value:dimensionValue];
```

Comment:
Your other idea also works, as optional params for `trackScreenView` and `trackEvent`. But the problem here being if we do that then those functions could grow a bit too big. 

Say someone wants to implement `setCustomMetric` later, I don't want that to be optional parameters as well on those methods. It seems cleaner to implemented a `setCustomDimensionWithScreen` and/or `setCustomDimensionWithEvent`instead.